hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieParquetInputFormat.java

```diff
@@ -91,9 +94,42 @@ private void initAvroInputFormat() {
     }
   }
 
+  private static boolean checkIfHudiTable(final InputSplit split, final JobConf job) {
+    try {
+      Option<Path> tablePathOpt = TablePathUtils.getTablePath(((FileSplit) split).getPath(), job);
+      if (!tablePathOpt.isPresent()) {
+        return false;
+      }
+      return tablePathOpt.get().getFileSystem(job).exists(new Path(tablePathOpt.get(), HoodieTableMetaClient.METAFOLDER_NAME));
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
   @Override
   public RecordReader<NullWritable, ArrayWritable> getRecordReader(final InputSplit split, final JobConf job,
                                                                    final Reporter reporter) throws IOException {
+
+    if (HoodieFileGroupReaderRecordReader.useFilegroupReader(job)) {
+      try {
```

Comment:
We need to confirm that the value of "hive.io.file.readcolumn.names" in Jobconf contains partition, if not, hive3 partition query returns null. see #7355

Reply:
https://github.com/apache/hudi/blob/2c38ef740d3d34e9eb05b59fa147c55623b81a90/hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieFileGroupReaderRecordReader.java#L249 I remove the partition fields from the read columns if the parquet file doesn't contain them. Does that help?

Reply:
This problem is parquet file in hive doesn't hava partition fields, while hudi parquet files have.